### ISSUE TITLE
[Feature]: Add a slimmed down reference implementation for nemotron H models

### ISSUE THREAD

**Issue by 2ez4bz** [NVIDIA/TensorRT-LLM#9230]
### 🚀 The feature, motivation and pitch

The reference nemotron H code on HuggingFace is out of date, and therefore bugged, and has several untested code paths. This makes an already hairy patching system even hairier.

The proposal is to do away with those patches, and replace the entire `AutoModelForCausalLM` implementation with one that is heavily slimmed down. Ideally, it should:

* work (fix bugs in the current implementation)
* add no new dependencies to TensorRT-LLM
* do away with unnecessary features for TensorRT-LLM / AutoDeploy:
  - no training related code (dropout, gradient checkpointing, etc.)
  - no caching business (we want to replace it with our own anyway)
  - no attention masking where possible
  - ...
* reuse existing AD custom ops for mamba SSM update / causal conv1d

If successful, this will also show an alternative path to the current patching approach.

### Alternatives

- The current patching approach.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.
